tests/components/brother/test_config_flow.py

```diff
@@ -116,12 +93,164 @@
 async def test_unsupported_model_error(hass):
     """Test unsupported printer model error."""
     with patch("brother.Brother._get_data", side_effect=UnsupportedModel("error")):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+        assert result["reason"] == "unsupported_model"
+
+
+async def test_device_exists_abort(hass):
+    """Test we abort config flow if Brother printer already configured."""
+    with patch(
+        "brother.Brother._get_data",
+        return_value=json.loads(load_fixture("brother_printer_data.json")),
+    ):
+        MockConfigEntry(domain=DOMAIN, unique_id="0123456789", data=CONFIG).add_to_hass(
+            hass
+        )
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+        assert result["reason"] == "already_configured"
+
+
+async def test_show_zeroconf_form(hass):
+    """Test that the zeroconf form is served."""
+    with patch(
+        "brother.Brother._get_data",
+        return_value=json.loads(load_fixture("brother_printer_data.json")),
+    ):
+        flow = config_flow.BrotherConfigFlow()
+        flow.hass = hass
+        flow.context = {"source": SOURCE_ZEROCONF}
+
+        result = await flow.async_step_zeroconf(
```

Comment:
Please use `hass.config_entries.flow.async_init` instead. We should always use the core interfaces to set up the tests.